### DISCUSSION TITLE
What is the most elegant way to integrate redis in a multi-level file project?

### DISCUSSION THREAD

**Discussion by axiangcoding** [fastapi/fastapi#9560] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python

```

### Description

What is the most elegant way to integrate redis in a multi-level file project? 

### Operating System

Windows

### FastAPI Version

0.95.2

### Python Version

3.11.3

**Comment by jgould22** [comment]
Redis is a database and you can/should treat it as such.

Initialize a Redis client on start up and inject it as a dependency in the same way you would pass in a database session.
